demokratie-live/democracy-server
# sprint#21/jwt

Implements Server-Side JWT authentication for users using headers

Also uses an IP-Whitelist for Bundestag.io Hooks

**Env**

Additional fields:
- DEBUG=false // Debug enabled mongolog & enables push-test/import-all hooks
- AUTH_JWT_TTL=1s  // Lifetime for Token
- AUTH_JWT_REFRESH_TTL=5s // Lifetime for Refreshtoken
- SMS_VERIFICATION=true // enables verified check (should be false for now)
- JWT_BACKWARD_COMPATIBILITY=false // Allows User.signUp mutation
- WHITELIST_DATA_SOURCES = IP,IP2 // Comma seperated list of whitelisted ips for bio-hooks

```
AUTH_JWT_SECRET=XXX
SECRET_KEY={XXX}
APPLE_TEAMID=XXX
APPLE_APN_KEY_ID=XXX
APPLE_APN_KEY=./XXX
NOTIFICATION_ANDROID_SERVER_KEY=XXX
STAGE=alpha
NODE_ENV=development
GRAPHIQL=true
HC_BACKEND_URL=XXX
HC_ORGANIZATION_SLUG=XXX
HC_LOGIN_EMAIL=XXX
HC_LOGIN_PASSWORD=XXX
DEBUG=true
AUTH_JWT_TTL=1s
AUTH_JWT_REFRESH_TTL=5s
SMS_VERIFICATION=false
JWT_BACKWARD_COMPATIBILITY=true
WHITELIST_DATA_SOURCES=::ffff:127.0.0.1,::1
```

**How2Test**

Client Test
- Checkout branch & Adjust .env
- Run Server and new Client
- Check if requests work like getlist, getdetails, addvote, addactivity, ...
- Enable SMS_VERIFICATAION and check if still voting works - should not
- Check out old Client (master), disable SMS_VERIFICATION
- Check if requests work like getlist, getdetails, addvote, addactivity, ...
- Disable JWT_BACKWARD_COMPATIBILITY and check if old client still works - should not

BIO Test
- Set WHITELIST_DATA_SOURCES = 111.111.111.111
- Run Server and BIO Server
- Wait for Hook - should throw Permission denied
- Set WHITELIST_DATA_SOURCES to your local IP (default value if localhost)(Connecting IP is shown in log)
- Run Server and BIO Server
- Wait for Hook - should be successfull now
